dnf5/commands/check/check.cpp

```diff
@@ -0,0 +1,336 @@
+    auto option = parser.add_new_named_arg(name);
+    option->set_long_name(name);
+    option->set_description(descr);
+    option->set_parse_hook_func([&opt_to_modify](cli::ArgumentParser::NamedArg *, const char *, const char *) {
+        opt_to_modify = true;
+        return true;
+    });
+    return option;
+}
+
+
+inline bool is_package_installonly(Header h) noexcept {
+    rpmtd td = rpmtdNew();
+    headerGet(h, RPMTAG_PROVIDES, td, 0);
+    while (auto provide = rpmtdNextString(td)) {
+        for (const auto & installonly_pkg_provide : installonly_pkgs_provides) {
+            if (installonly_pkg_provide == provide) {
+                rpmtdFree(td);
+                return true;
+            }
+        }
+    }
+    rpmtdFree(td);
+
+    return false;
+}
+
+
+// Ensures that NEVRA always contains an EPOCH.
+// Returns the input `nevra`. EPOCH is inserted if it was not on the input.
+// librpm functions such as "rpmProblemGetPkgNEVR()" and "rpmProblemGetAltNEVR()" return epoch 0 only sometimes.
+// We want the epoch always.
+inline std::string ensure_full_nevra(const char * nevra) {
+    auto nevras = rpm::Nevra::parse(nevra, {rpm::Nevra::Form::NEVRA});
+    return rpm::to_full_nevra_string(nevras.front());
+}
+
+
+// Finds problems for the installed package defined by Header h.
+// Principle: Adds the installed package to an empty transaction and checks for errors.
+int find_problems(QVA_t qva, rpmts ts, Header h) {
+    int rc = 0;
+
+    if (qva->qva_flags & VERIFY_DEPS) {
+        rpmtsEmpty(ts);
+        (void)rpmtsAddInstallElement(ts, h, NULL, 0, NULL);
+
+        (void)rpmtsCheck(ts);
+        auto te = rpmtsElement(ts, 0);
+        auto ps = rpmteProblems(te);
+        rc = rpmpsNumProblems(ps);
+
+        if (rc > 0) {
+            rpmpsi psi = rpmpsInitIterator(ps);
+            while (auto p = rpmpsiNext(psi)) {
+                auto problem_cstr = rpmProblemGetStr(p);
+                libdnf_assert(problem_cstr != NULL, "command \"check\": rpmProblemGetStr() returns NULL");
+                auto cnevr = rpmProblemGetPkgNEVR(p);
+                libdnf_assert(cnevr != NULL, "command \"check\": rpmProblemGetPkgNEVR() returns NULL");
+                auto alt_cnevr = rpmProblemGetAltNEVR(p);
+                libdnf_assert(alt_cnevr != NULL, "command \"check\": rpmProblemGetAltNEVR() returns NULL");
+
+                switch (rpmProblemGetType(p)) {
+                    case RPMPROB_REQUIRES:
+                        if (all || dependencies) {
+                            problems[ensure_full_nevra(alt_cnevr)].insert(Problem{
+                                .type = ProblemType::MISSING_REQUIRE, .nevra = "", .file_or_provide = problem_cstr});
+                        }
+                        break;
+                    case RPMPROB_CONFLICT:
+                        if (all || dependencies) {
+                            auto nevr = ensure_full_nevra(cnevr);
```

Comment:
After our discussion it was reveals that `nevr` variable contains full nevra string including EPOCH. May I ask you for renaming the variable?